### PR TITLE
chore(docs): Update "Sourcing from Kontent.ai"

### PR DESCRIPTION
**Pull request by IvanKiral**
## Description

Updated md file of the documentation 'Sourcing from Kontent.ai'. Most of the changes include rebranding, changing links and updating code snippets

### CODE REVIEW

**Comment by LekoArts** [comment]
Hi, thanks for the PR!

Nowadays we like to link to the guides that CMSs put up and then just link to it, as seen on https://www.gatsbyjs.com/docs/how-to/sourcing-data/headless-cms/. But I couldn't find a doc on https://kontent.ai/learn/tutorials/develop-apps/optimize-your-app/static-sites/?tech=react so I guess for now we can keep the guide.

Before reviewing the actual contents of the guide, please move the file to this location:

```shell
gatsby/docs/docs/how-to/sourcing-data
```

And name the file `sourcing-from-kontent-ai.md`

**Comment by IvanKiral** [comment]
Thanks. I have updated the directory of the doc :)

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+Once you have a subscription to Kontent.ai, you need some content to retrieve. If you know what you want, you can set up your own content types (templates for your content) and then create content items (the actual content) based on them. If you'd like to take a shortcut and see some example content, you can use [Sample Project generator](https://app.kontent.ai/sample-project-generator) to generate project "Sample Project" and import sample content. This guide will continue with the example of the "Sample Project".
+
+The created Sample Project is a comprehensive presentation of a fictional coffee company named Dancing Goat that showcases various Kontent.ai features. It can be displayed in various channels, as you can see by going to the [Quickstart page in Kontent.ai](https://app.kontent.ai/quickstart) from within that project.
+
+For this guide, you don't have to worry about most of the features. You'll pull some data to display within the Gatsby site you'll create in the next step. The only thing you need to get is your Project ID, which you can find in Kontent.ai under _Project settings_ -> _API keys_.
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:23]:
```suggestion
For this guide, you don't have to worry about most of the features. You'll pull some data to display within the Gatsby site you'll create in the next step. The only thing you need to get is your Project ID, which you can find in Kontent.ai under **"Project settings"** -> **"API keys"**.
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+```shell
+gatsby new kontent-guide
+cd kontent-guide
+```
+
+Next, install the [Kontent.ai source plugin](https://github.com/kontent-ai/gatsby-packages/tree/master/packages/gatsby-source):
+
+```shell
+npm install --save @kontent-ai/gatsby-source
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:39]:
```suggestion
npm install @kontent-ai/gatsby-source
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+```
+
+Once that's done, you need to add the plugin to `gatsby-config.js`:
+
+```javascript
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:44]:
```suggestion
```js:title=gatsby-config.js
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+cd kontent-guide
+```
+
+Next, install the [Kontent.ai source plugin](https://github.com/kontent-ai/gatsby-packages/tree/master/packages/gatsby-source):
+
+```shell
+npm install --save @kontent-ai/gatsby-source
+```
+
+Once that's done, you need to add the plugin to `gatsby-config.js`:
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:42]:
```suggestion
Once that's done, you need to add the plugin to `gatsby-config`:
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+}
+```
+
+And that's enough for you to be able to access content from Kontent.ai in your site. You can see this by starting Gatsby in development mode:
+
+```shell
+gatsby develop
+```
+
+To see all the content that's available from Kontent.ai, you can test out GraphQL queries in GraphiQL at `http://localhost:8000/___graphql`. The queries generated from Kontent.ai will be prefixed with `kontentItem` (for single nodes) or `allKontentItem`.
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:72]:
```suggestion
To see all the content that's available from Kontent.ai, you can test out GraphQL queries in [GraphiQL](/docs/how-to/querying-data/running-queries-with-graphiql/) at `http://localhost:8000/___graphql`. The queries generated from Kontent.ai will be prefixed with `kontentItem` (for single nodes) or `allKontentItem`.
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+Once you have a subscription to Kontent.ai, you need some content to retrieve. If you know what you want, you can set up your own content types (templates for your content) and then create content items (the actual content) based on them. If you'd like to take a shortcut and see some example content, you can use [Sample Project generator](https://app.kontent.ai/sample-project-generator) to generate project "Sample Project" and import sample content. This guide will continue with the example of the "Sample Project".
+
+The created Sample Project is a comprehensive presentation of a fictional coffee company named Dancing Goat that showcases various Kontent.ai features. It can be displayed in various channels, as you can see by going to the [Quickstart page in Kontent.ai](https://app.kontent.ai/quickstart) from within that project.
+
+For this guide, you don't have to worry about most of the features. You'll pull some data to display within the Gatsby site you'll create in the next step. The only thing you need to get is your Project ID, which you can find in Kontent.ai under _Project settings_ -> _API keys_.
+
+### Gatsby
+
+#### Adding content to existing pages
+
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:28]:
```suggestion
### Gatsby

```

The setup would only be adding the plugin. The "adding content" piece would be under a new h2 heading (not "setup")

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+And that's enough for you to be able to access content from Kontent.ai in your site. You can see this by starting Gatsby in development mode:
+
+```shell
+gatsby develop
+```
+
+To see all the content that's available from Kontent.ai, you can test out GraphQL queries in GraphiQL at `http://localhost:8000/___graphql`. The queries generated from Kontent.ai will be prefixed with `kontentItem` (for single nodes) or `allKontentItem`.
+
+To see how to put that data into your site, first go to `http://localhost:8000/`. Notice that the default title for the site is "Gatsby Default Starter". You can change that by pulling the title for your site from Kontent.ai.
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:74]:
```suggestion
## Using the plugin

### Adding content to existing pages

To see how to put that data into your site, first go to `http://localhost:8000/`. Notice that the default title for the site is "Gatsby Default Starter". You can change that by pulling the title for your site from Kontent.ai.
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+    <>
+      <Header siteTitle={data.kontentItemHome.elements.metadata__meta_title.value} />
+// ...
+```
+
+If you look at `http://localhost:8000/`, you'll notice the title is now "Dancing Goat–Freshest coffee on the block!". You can change this title in Kontent.ai to whatever you want and rerun `gatsby develop` to rebuild the site ([see below about automatic builds](#continuous-deployment)).
+
+So you've seen how to add content to existing pages in Gatsby using Kontent.ai. Next, you will start creating new pages of your own.
+
+#### Creating new pages
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:103]:
```suggestion
### Creating new pages
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+
+It's great to be able to add content to existing static pages, but one of the great benefits of using CaaS is being able to define pages in Kontent.ai and having them generated automatically. To see how, you'll add pages based on content from the Sample Project in the Article type (feel free to explore how these are structured in Kontent.ai).
+
+Start by making use of the URL pattern in the Article type to generate slugs for your Article nodes:
+
+```javascript:title=gatsby-node.js
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:109]:
```suggestion
```js:title=gatsby-node.js
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+      name: `slug`,
+      value: node.elements.url_pattern.value,
+    })
+  }
+}
+```
+
+Now that you have a pretty way to define the path for your pages, you can create the pages programmatically:
+
+```javascript:title=gatsby-node.js
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:123]:
```suggestion
```js:title=gatsby-node.js
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+  const result = await graphql(`
+    {
+      allKontentItemArticle {
+        edges {
+          node {
+            fields {
+              slug
+            }
+          }
+        }
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:150]:
```suggestion
        nodes {
          fields {
            slug
          }
        }
```

commonly used syntax now

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+      }
+    }
+  `)
+
+  // Create pages
+  result.data.allKontentItemArticle.edges.forEach(({ node }) => {
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:156]:
```suggestion
  result.data.allKontentItemArticle.nodes.forEach((node) => {
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+
+Since the Kontent.ai source plugin is defining the GraphQL schema for data from Kontent.ai, you could use this schema and extend it according to your needs. There are [some examples](https://github.com/kontent-ai/gatsby-packages/tree/master/site#examples) of what you could do in your application.
+
+Now you know how to create pages programmatically and pull their content from Kontent.ai. To get the most out of your CaaS, you'll want to also make sure your site builds automatically whenever published content changes inside Kontent.ai.
+
+### Continuous deployment
+
+To keep your site static but always up to date with the latest content from Kontent.ai, it helps to set up automatic deployment whenever your published content changes. Here, you can see how to set that up using [Netlify](https://docs.netlify.com/site-deploys/create-deploys/#deploy-with-git), but the principle is similar using other services like [Gatsby Cloud](https://support.gatsbyjs.com/hc/en-us/articles/360052324654-Connecting-to-Kontent), or [Travis CI](https://travis-ci.org/), etc...
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:219]:
```suggestion
To keep your site static but always up to date with the latest content from Kontent.ai, it helps to set up automatic deployment whenever your published content changes. Reade the [Gatsby Cloud](https://support.gatsbyjs.com/hc/en-us/articles/360052324654-Connecting-to-Kontent) instructions to learn more.
```

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+
+Now that your site's up and running, you need to set up automatic builds when published content in Kontent.ai changes. First, in Netlify [create a new build hook](https://docs.netlify.com/configure-builds/build-hooks/) with a name like "Change in Kontent.ai content" and copy the URL. Then go to Kontent.ai. Under _Project settings_, choose _Webhooks_ and create a new webhook. Give it a name like "Netlify build", paste the URL into the _URL address_ field and choose the [events to trigger the webhook](https://docs.kontent.ai/reference/webhooks-reference#a-events-to-trigger-a-webhook), you want to select just "DELIVERY API TRIGGERS" for content item events: "Publish" and "Unpublish". And that's it. Now whenever published content changes, your webhook will trigger a build in Netlify to ensure your static content is updated to the latest version.
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:223]:
```suggestion
For automatic deployment from Netlify you'll want to [create a new build hook](https://docs.netlify.com/configure-builds/build-hooks/) with a name like "Change in Kontent.ai content" and copy the URL. Then go to Kontent.ai. Under **"Project settings"**, choose **"Webhooks"** and create a new webhook. Give it a name like "Netlify build", paste the URL into the **"URL address"** field and choose the [events to trigger the webhook](https://docs.kontent.ai/reference/webhooks-reference#a-events-to-trigger-a-webhook), you want to select just "DELIVERY API TRIGGERS" for content item events: "Publish" and "Unpublish". And that's it. Now whenever published content changes, your webhook will trigger a build in Netlify to ensure your static content is updated to the latest version.
```

**Review by LekoArts — CHANGES_REQUESTED** [review]

File: docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md (hunk 3 above)
Comment (IvanKiral) [reply docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:44]: What is this doing?

Comment (LekoArts) [reply docs/docs/how-to/sourcing-data/sourcing-from-kontent-ai.md:42]: https://www.gatsbyjs.com/contributing/gatsby-style-guide/#code-formatting-titles

**Review by LekoArts — APPROVED** [review]

**Pull request merged**